exchanges/gateio/gateio_websocket.go

```diff
@@ -581,32 +584,34 @@ func (g *Gateio) manageSubs(ctx context.Context, event string, conn stream.Conne
 		return errs
 	}
 
-	for _, s := range subs {
-		if err := func() error {
-			msg, err := g.manageSubReq(ctx, event, conn, s)
-			if err != nil {
-				return err
-			}
-			result, err := conn.SendMessageReturnResponse(ctx, request.Unset, msg.ID, msg)
-			if err != nil {
-				return err
-			}
-			var resp WsEventResponse
-			if err := json.Unmarshal(result, &resp); err != nil {
-				return err
-			}
-			if resp.Error != nil && resp.Error.Code != 0 {
-				return fmt.Errorf("(%d) %s", resp.Error.Code, resp.Error.Message)
-			}
-			if event == "unsubscribe" {
-				return g.Websocket.RemoveSubscriptions(conn, s)
-			}
-			return g.Websocket.AddSuccessfulSubscriptions(conn, s)
-		}(); err != nil {
-			errs = common.AppendError(errs, fmt.Errorf("%s %s %s: %w", s.Channel, s.Asset, s.Pairs, err))
+	return common.ProcessElementsByBatch(subscriptionBatchCount, subs, func(_ int, s *subscription.Subscription) error {
+		if err := g.manageTemplatePayload(ctx, conn, event, s); err != nil {
+			return fmt.Errorf("%s %s %s: %w", s.Channel, s.Asset, s.Pairs, err)
 		}
+		return nil
+	})
+}
+
+func (g *Gateio) manageTemplatePayload(ctx context.Context, conn stream.Connection, event string, s *subscription.Subscription) error {
```

Comment:
This isn't anything to do with templates, especially since a user might have called Subscribe manually.
I think this is:
```suggestion
func (g *Gateio) manageSubPayload(ctx context.Context, conn stream.Connection, event string, s *subscription.Subscription) error {
```

Reply:
Think you've made Subscribe into manageSubs, and manageSubs into manageTemplatePayload :)